commit 524a8ffc7efbc10b09234f7ac3f81d30a3f9f65c
Author: Bernhard M. Wiedemann <bwiedemann@suse.de>
Date:   Sun Apr 30 22:47:40 2017 +0200

    created.rid: use SOURCE_DATE_EPOCH
    
    use SOURCE_DATE_EPOCH instead of current time in created.rid top line
    to enable reproducible builds of ruby docs
    
    See https://reproducible-builds.org/ for why this is good
    and https://reproducible-builds.org/specs/source-date-epoch/
    for the definition of this variable.

diff --git a/lib/rdoc/rdoc.rb b/lib/rdoc/rdoc.rb
index 68775c8b..a2711fbb 100644
--- a/lib/rdoc/rdoc.rb
+++ b/lib/rdoc/rdoc.rb
@@ -232,6 +232,9 @@ option)
 
   def update_output_dir(op_dir, time, last = {})
     return if @options.dry_run or not @options.update_output_dir
+    unless ENV['SOURCE_DATE_EPOCH'].nil?
+      time = Time.at(ENV['SOURCE_DATE_EPOCH'].to_i).gmtime
+    end
 
     open output_flag_file(op_dir), "w" do |f|
       f.puts time.rfc2822
